src/test/java/MacroExecutionModuleImplementationTest.java

```diff
@@ -0,0 +1,187 @@
+        mockCloudDocClient = Mockito.mock(CloudDocClient.class);
+        mockCloudSheet = Mockito.mock(CloudSheet.class);
+        mockDriveCloudSheet = Mockito.mock(DriveCloudSheet.class);
+        mockAction = Mockito.mock(Action.class);    
+        execution  = MacroExecutionModuleImplementation.initializeServer(mockEntityModule, mockCloudDocClient);
+    }
+    
+    // Remove a macro from the roomToMacro Map
+    @Test
+    public void testRemoveMacro() {
+        Map<String, Map<String, String>> actual = execution.getRoomToMacro();
+        Map<String, String> macro = new HashMap<String, String>();
+        macro.put(MACRO_NAME, USER_EMAIL);
+        actual.put(ROOM1, macro);
+
+        execution.removeMacro(ROOM1);
+
+        Map<String, Map<String, String>> expected = new HashMap<String, Map<String, String>>();
+        Assert.assertEquals(expected, actual);
+    }
+
+    // Get macro name from the message when the macro has not yet been used in the thread
+    @Test
+    public void getMacroNameFromMessage() {
+        emptyThreadMacroMap();
+        String actual = execution.getMacroName(MESSAGE_WITH_MACRO_NAME, THREAD1);
+        Assert.assertEquals(MACRO_NAME, actual);
+    }
+
+    // Get macro name from the message when the macro has already been used in the thread (ie macro name is not stated explicitly in the message)
+    @Test
+    public void getMacroNameFromThread() {
+        addToThreadMacroMap(THREAD1, MACRO_NAME);
+        String actual = execution.getMacroName(MESSAGE_WITHOUT_MACRO_NAME, THREAD1);
+        Assert.assertEquals(MACRO_NAME, actual);
+    }
+
+    // Test the help message
+    @Test
+    public void testHelpMessage() throws IOException, GeneralSecurityException {
+        ChatResponse chatResponse = execution.getReplyText("@MacroBot /help", THREAD1, ROOM1, USER_EMAIL, HELP_MESSAGE);
+        String actual = chatResponse.getReplyText();
+        Assert.assertEquals(HELP_MESSAGE, actual);
+    }
+
+    // Test an empty message
+    @Test
+    public void testEmptyMessage() throws IOException, GeneralSecurityException {
+        ChatResponse chatResponse = execution.getReplyText("@MacroBot", THREAD1, ROOM1, USER_EMAIL, HELP_MESSAGE);
+        String actual = chatResponse.getReplyText();
+        Assert.assertEquals(EMPTY_MESSAGE_RESPONSE, actual);
+    }
+
+    // Test using a macro for the first time when you are the creator. Testing with SheetAppendRow as action type. 
+    @Test
+    public void useYourMacroInNewRoom() throws IOException, GeneralSecurityException, Exception {
+        List<String> values = new ArrayList<String>();
+        values.add(getDate("dd-MM-yyyy HH:mm:ss"));
+        values.add(USER_EMAIL);
+        //TODO: change this message to use a var instead of hardcoded
+        values.add("please log my message");
+        
+        when(mockEntityModule.getMacro(USER_EMAIL, MACRO_NAME)).thenReturn(Optional.of(DEFAULT_MACRO));
```

Comment:
The null pointer exception comes from the fact that DEFAULT_MACRO is disconnected from mockAction. To make it work you would have to somehow add the mockAction object to the DEFAULT_MACRO object. But there is an easier and better approach. Simply initialize DEFAULT_MACRO with proper sheetUrl. As you can see the documentId comes from the documentUrl [1]. If I understand the regex correctly making sheetUrl "d/sheetId" should be enough to make things working.

[1] https://github.com/googleinterns/step81-2020/blob/master/src/main/java/com/google/flourbot/entity/action/sheet/SheetAction.java